crates/ty_python_semantic/src/semantic_index/builder.rs

```diff
@@ -550,6 +550,25 @@ impl<'db, 'ast> SemanticIndexBuilder<'db, 'ast> {
             match node {
                 ast::Expr::BooleanLiteral(ast::ExprBooleanLiteral { value, .. }) => Some(*value),
                 ast::Expr::Name(ast::ExprName { id, .. }) if id == "TYPE_CHECKING" => Some(true),
+                ast::Expr::NumberLiteral(ast::ExprNumberLiteral { value, .. }) => match value {
+                    ast::Number::Int(n) => Some(*n != 0),
+                    ast::Number::Float(n) => Some(*n != 0.0),
+                    ast::Number::Complex { real, imag } => Some(*real != 0.0 && *imag != 0.0),
+                },
+                ast::Expr::StringLiteral(ast::ExprStringLiteral { value, .. }) => {
+                    Some(!value.is_empty())
+                }
+                ast::Expr::BytesLiteral(ast::ExprBytesLiteral { value, .. }) => {
+                    Some(!value.is_empty())
+                }
+                ast::Expr::FString(ast::ExprFString { value, .. }) => {
+                    Some(!value.is_empty_literal())
+                }
+                ast::Expr::Tuple(ast::ExprTuple { elts, .. }) => Some(!elts.is_empty()),
+                ast::Expr::EllipsisLiteral(_) | ast::Expr::Lambda(_) | ast::Expr::Generator(_) => {
+                    Some(true)
+                }
+                ast::Expr::NoneLiteral(_) => Some(false),
```

Comment:
Should we limit the complexity and the potential inconsistency between this and type inference here? For example, whether or not empty tuples should be considered always-falsy was a recent matter of debate. And it seems extremely unlikely that someone would use something like `if f""` of `if 0j`.

I would personally say the `ast::Number::Int(_)` change and `ast::Expr::NoneLiteral` seem like reasonable additions. Everything else, I would remove again. It only adds code complexity, and we do perform additional work for no gain (if those patterns are never used).

Reply:
make sense, thanks!